Fix Gradle daemon crashes on CircleCI

The message received from the daemon indicates that the daemon has disappeared.
Build request sent: Build{id=f57a1c2e-cec7-42d1-9bb5-a5039f6179b6, currentDir=/home/circleci/code}
Attempting to read last messages from the daemon log...
Daemon pid: 125
  log file: /home/circleci/.gradle/daemon/6.3/daemon-125.out.log
----- Last  20 lines from daemon log file - daemon-125.out.log -----
2020-05-10T13:51:18.982+0000 [INFO] [org.gradle.launcher.daemon.server.DefaultIncomingConnectionHandler] Received command: Build{id=f57a1c2e-cec7-42d1-9bb5-a5039f6179b6, currentDir=/home/circleci/code}.
2020-05-10T13:51:18.983+0000 [DEBUG] [org.gradle.launcher.daemon.server.DefaultIncomingConnectionHandler] Starting executing command: Build{id=f57a1c2e-cec7-42d1-9bb5-a5039f6179b6, currentDir=/home/circleci/code} with connection: socket connection from /127.0.0.1:45275 to /127.0.0.1:47566.
2020-05-10T13:51:18.983+0000 [ERROR] [org.gradle.launcher.daemon.server.DaemonStateCoordinator] Command execution: started DaemonCommandExecution[command = Build{id=f57a1c2e-cec7-42d1-9bb5-a5039f6179b6, currentDir=/home/circleci/code}, connection = DefaultDaemonConnection: socket connection from /127.0.0.1:45275 to /127.0.0.1:47566] after 0.0 minutes of idle
2020-05-10T13:51:18.983+0000 [INFO] [org.gradle.launcher.daemon.server.DaemonRegistryUpdater] Marking the daemon as busy, address: [a6c0368a-58cc-41d9-aeed-95ddb52428e5 port:45275, addresses:[localhost/127.0.0.1]]
2020-05-10T13:51:18.983+0000 [DEBUG] [org.gradle.launcher.daemon.registry.PersistentDaemonRegistry] Marking busy by address: [a6c0368a-58cc-41d9-aeed-95ddb52428e5 port:45275, addresses:[localhost/127.0.0.1]]
2020-05-10T13:51:18.983+0000 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire exclusive lock on daemon addresses registry.
2020-05-10T13:51:18.984+0000 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
2020-05-10T13:51:18.984+0000 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
2020-05-10T13:51:18.985+0000 [DEBUG] [org.gradle.launcher.daemon.server.DaemonStateCoordinator] resetting idle timer
2020-05-10T13:51:18.985+0000 [DEBUG] [org.gradle.launcher.daemon.server.DaemonStateCoordinator] daemon is running. Sleeping until state changes.
2020-05-10T13:51:18.985+0000 [INFO] [org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy] Daemon is about to start building Build{id=f57a1c2e-cec7-42d1-9bb5-a5039f6179b6, currentDir=/home/circleci/code}. Dispatching build started information...
2020-05-10T13:51:18.985+0000 [DEBUG] [org.gradle.launcher.daemon.server.SynchronizedDispatchConnection] thread 30: dispatching org.gradle.launcher.daemon.protocol.BuildStarted@6e6cec9f
2020-05-10T13:51:18.985+0000 [DEBUG] [org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment] Configuring env variables: {PATH=/opt/android/sdk/emulator:/opt/android/sdk/tools:/opt/android/sdk/tools/bin:/opt/android/sdk/platform-tools:/home/circleci/.rubies/ruby-2.6.1/bin:/home/circleci/.local/bin:/home/circleci/bin:/usr/local/openjdk-8/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin, bintrayuser=***************, NO_PROXY=127.0.0.1,localhost,circleci-internal-outer-build-agent, JAVA_URL_VERSION=8u252b09, GRADLE_OPTS=-Xmx1536m -XX:+HeapDumpOnOutOfMemoryError -Dorg.gradle.caching=true -Dorg.gradle.configureondemand=true -Dkotlin.compiler.execution.strategy=in-process -Dkotlin.incremental=false, CIRCLE_WORKFLOW_WORKSPACE_ID=132754eb-9f13-412b-87bd-48845357e8dd, CIRCLE_PREVIOUS_BUILD_NUM=224, CIRCLE_PULL_REQUEST=https://github.com/KasperskyLab/Kaspresso/pull/142, ANDROID_HOME=/opt/android/sdk, CIRCLE_PROJECT_REPONAME=Kaspresso, artifactoryPassword=**********************************************************************, CIRCLE_WORKING_DIRECTORY=~/code, CIRCLE_INTERNAL_TASK_DATA=/.circleci-task-data, PWD=/home/circleci/code, CI_PULL_REQUESTS=https://github.com/KasperskyLab/Kaspresso/pull/142, BASH_ENV=/tmp/.bash_env-5eb805f79689ec089eb760ad-0-build, CIRCLE_BUILD_NUM=225, CI_PULL_REQUEST=https://github.com/KasperskyLab/Kaspresso/pull/142, JAVA_BASE_URL=https://github.com/AdoptOpenJDK/openjdk8-upstream-binaries/releases/download/jdk8u252-b09/OpenJDK8U-jdk_, CIRCLE_SHA1=ddb6677a86969062aae93fc1134e3c5cdfb3cc6e, OLDPWD=/home/circleci/code, CIRCLE_NODE_INDEX=0, JVM_OPTS=-Xmx4g, ADB_INSTALL_TIMEOUT=120, CIRCLE_NODE_TOTAL=1, DEBIAN_FRONTEND=noninteractive, CIRCLE_SHELL_ENV=/tmp/.bash_env-5eb805f79689ec089eb760ad-0-build, SHLVL=0, JAVA_HOME=/usr/local/openjdk-8, CIRCLE_COMPARE_URL=https://github.com/KasperskyLab/Kaspresso/compare/75f8540156c6f778c38e637ae0a889ecc9e602d4...ddb6677a86969062aae93fc1134e3c5cdfb3cc6e, CIRCLE_PROJECT_USERNAME=KasperskyLab, LANG=C.UTF-8, CIRCLE_STAGE=analyze, CIRCLE_INTERNAL_SCRATCH=/tmp/circleci-186637445, CIRCLE_BUILD_URL=https://circleci.com/gh/KasperskyLab/Kaspresso/225, CIRCLE_WORKFLOW_UPSTREAM_JOB_IDS=, CIRCLE_BRANCH=issue/109, CIRCLE_PULL_REQUESTS=https://github.com/KasperskyLab/Kaspresso/pull/142, bintraykey=****************************************, JAVA_VERSION=8u252, _=./gradlew, CIRCLECI=true, CIRCLE_USERNAME=matzuk, CIRCLE_REPOSITORY_URL=git@github.com:KasperskyLab/Kaspresso.git, CI=true, CIRCLE_JOB=analyze, CIRCLE_WORKFLOW_ID=132754eb-9f13-412b-87bd-48845357e8dd, SSH_AUTH_SOCK=/tmp/circleci-186637445/ssh_auth_sock, CIRCLE_INTERNAL_CONFIG=/.circleci-runner-config.json, HOSTNAME=bd039a4cf744, CIRCLE_WORKFLOW_JOB_ID=877ffff3-2e2b-4c27-8667-c59e9dab986a, HOME=/home/circleci}
2020-05-10T13:51:18.985+0000 [DEBUG] [org.gradle.launcher.daemon.server.exec.LogToClient] About to start relaying all logs to the client via the connection.
2020-05-10T13:51:18.985+0000 [INFO] [org.gradle.launcher.daemon.server.exec.LogToClient] The client will now receive all logging from the daemon (pid: 125). The daemon log file: /home/circleci/.gradle/daemon/6.3/daemon-125.out.log
2020-05-10T13:51:18.986+0000 [INFO] [org.gradle.launcher.daemon.server.exec.LogAndCheckHealth] Starting 4th build in daemon [uptime: 3 mins 28.847 secs, performance: 97%, GC rate: 0.07/s, heap usage: 7% of 1.4 GB]
2020-05-10T13:51:18.986+0000 [DEBUG] [org.gradle.launcher.daemon.server.exec.ExecuteBuild] The daemon has started executing the build.
2020-05-10T13:51:18.986+0000 [DEBUG] [org.gradle.launcher.daemon.server.exec.ExecuteBuild] Executing build with daemon context: DefaultDaemonContext[uid=cd0974ad-4c37-4833-b295-ea2c9f1ec368,javaHome=/usr/local/openjdk-8,daemonRegistryDir=/home/circleci/.gradle/daemon,pid=125,idleTimeout=10800000,priority=NORMAL,daemonOpts=-XX:MaxPermSize=512m,-XX:+HeapDumpOnOutOfMemoryError,-Xmx2048m,-Dfile.encoding=UTF-8,-Duser.country,-Duser.language=en,-Duser.variant]
Configuration on demand is an incubating feature.
AAPT2 aapt2-3.6.3-6040484-linux Daemon #8: Idle daemon unexpectedly exit. This should not happen.
----- End of the daemon log -----

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -6,7 +6,7 @@ jobs:
       - image: circleci/android:api-29
     environment:
       JVM_OPTS: -Xmx4g
-      GRADLE_OPTS: -Xmx1536m -XX:+HeapDumpOnOutOfMemoryError -Dorg.gradle.caching=true -Dorg.gradle.configureondemand=true -Dkotlin.compiler.execution.strategy=in-process -Dkotlin.incremental=false
+      GRADLE_OPTS: --no-daemon -Xmx1536m -XX:+HeapDumpOnOutOfMemoryError -Dorg.gradle.caching=true -Dorg.gradle.configureondemand=true -Dkotlin.compiler.execution.strategy=in-process -Dkotlin.incremental=false
     steps:
       - checkout
       - run:
@@ -41,7 +41,7 @@ jobs:
       - image: circleci/android:api-29
     environment:
       JVM_OPTS: -Xmx4g
-      GRADLE_OPTS: -Xmx1536m -XX:+HeapDumpOnOutOfMemoryError -Dorg.gradle.caching=true -Dorg.gradle.configureondemand=true -Dkotlin.compiler.execution.strategy=in-process -Dkotlin.incremental=false
+      GRADLE_OPTS: --no-daemon -Xmx1536m -XX:+HeapDumpOnOutOfMemoryError -Dorg.gradle.caching=true -Dorg.gradle.configureondemand=true -Dkotlin.compiler.execution.strategy=in-process -Dkotlin.incremental=false
     steps:
       - checkout
       - run:

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -6,7 +6,7 @@ jobs:
       - image: circleci/android:api-29
     environment:
       JVM_OPTS: -Xmx4g
-      GRADLE_OPTS: --no-daemon -Xmx1536m -XX:+HeapDumpOnOutOfMemoryError -Dorg.gradle.caching=true -Dorg.gradle.configureondemand=true -Dkotlin.compiler.execution.strategy=in-process -Dkotlin.incremental=false
+      GRADLE_OPTS: -Dorg.gradle.daemon=false -Xmx1536m -XX:+HeapDumpOnOutOfMemoryError -Dorg.gradle.caching=true -Dorg.gradle.configureondemand=true -Dkotlin.compiler.execution.strategy=in-process -Dkotlin.incremental=false
     steps:
       - checkout
       - run:
@@ -41,7 +41,7 @@ jobs:
       - image: circleci/android:api-29
     environment:
       JVM_OPTS: -Xmx4g
-      GRADLE_OPTS: --no-daemon -Xmx1536m -XX:+HeapDumpOnOutOfMemoryError -Dorg.gradle.caching=true -Dorg.gradle.configureondemand=true -Dkotlin.compiler.execution.strategy=in-process -Dkotlin.incremental=false
+      GRADLE_OPTS: -Dorg.gradle.daemon=false -Xmx1536m -XX:+HeapDumpOnOutOfMemoryError -Dorg.gradle.caching=true -Dorg.gradle.configureondemand=true -Dkotlin.compiler.execution.strategy=in-process -Dkotlin.incremental=false
     steps:
       - checkout
       - run:

